rest/server_context_test.go

```diff
@@ -163,3 +165,197 @@ func (client *MockClient) RespondToGET(url string, response *http.Response) {
+		assert.Equal(t, bucketName, dbContext.BucketSpec.BucketName)
+		assert.ElementsMatch(t, databaseNames, serverContext.AllDatabaseNames())
+	}
+
+	databaseNames = removeElementWithValue(databaseNames, "imdb2")
+	databaseNames = removeElementWithValue(databaseNames, "imdb3")
+
+	status := serverContext.RemoveDatabase("imdb2")
+	assert.True(t, status, "Database should be removed from server context")
+	status = serverContext.RemoveDatabase("imdb3")
+	assert.True(t, status, "Database should be removed from server context")
+
+	assert.ElementsMatch(t, databaseNames, serverContext.AllDatabaseNames())
+}
+
+func removeElementWithIndex(s []string, index int) []string {
+	return append(s[:index], s[index+1:]...)
+}
+
+func removeElementWithValue(s []string, e string) []string {
+	return removeElementWithIndex(s, indexOf(s, e))
+}
+
+func indexOf(elements []string, e string) int {
+	for k, v := range elements {
+		if e == v {
+			return k
+		}
+	}
+	return -1 // Not found.
+}
+
+func captureOutput(f func()) string {
+	var buf bytes.Buffer
+	log.SetOutput(&buf)
+	defer func() { log.SetOutput(os.Stderr) }()
+	f()
+	return buf.String()
+}
+
+func TestStartReplicators(t *testing.T) {
+	defer base.SetUpTestLogging(base.LevelInfo, base.KeyReplicate)()
+	var replications []*ReplicationConfig
+
+	// Should be skipped; create_target option is not currently supported.
+	replications = append(replications, &ReplicationConfig{
+		Source:       "http://127.0.0.1:4985/imdb_us",
+		Target:       "http://127.0.0.1:4985/imdb_uk",
+		Async:        true,
+		CreateTarget: true})
+
+	// Should be skipped; doc_ids option is not currently supported
+	replications = append(replications, &ReplicationConfig{
+		Source: "http://127.0.0.1:4985/imdb_us",
+		Target: "http://127.0.0.1:4985/imdb_uk",
+		Async:  true,
+		DocIds: []string{"doc1", "doc2", "doc3", "doc4"}})
+
+	// Should be skipped; proxy option is not currently supported.
+	replications = append(replications, &ReplicationConfig{
+		Source: "http://127.0.0.1:4985/imdb_us",
+		Target: "http://127.0.0.1:4985/imdb_uk",
+		Async:  true,
+		Proxy:  "http://127.0.0.1:443/proxy"})
+
+	// Start a one-shot replication.
+	replications = append(replications, &ReplicationConfig{
+		Source:        "http://127.0.0.1:4985/imdb_us",
+		Target:        "http://127.0.0.1:4985/imdb_uk",
+		Async:         true,
+		ReplicationId: "58a632bb8d7e110445d3d65a98365d62"})
+
+	serverConfig := &ServerConfig{AdminInterface: &DefaultAdminInterface, Replications: replications}
+	serverContext := NewServerContext(serverConfig)
+	out := captureOutput(serverContext.startReplicators)
+
+	assert.Contains(t, "/_replicate create_target option is not currently supported", out)
+	assert.Contains(t, "/_replicate doc_ids option is not currently supported", out)
+	assert.Contains(t, "/_replicate proxy option is not currently supported", out)
+	assert.Contains(t, "ReplicationId:58a632bb8d7e110445d3d65a98365d62", out)
```

Comment:
This seems a little unreliable, and probably not the right direction for a few reasons:
- We're relying on the logging not changing
- We're relying on log levels/log keys to be set properly for the test assertions to work
- as we're only asserting the logs for the entire test run, and can't really associate a particular error with a replication above

What test coverage are we aiming for here with this test? It seems like we are better able to test various error scenarios in `validateReplicationParameters` with the `handleReplicate` endpoint.

Reply:
Makes sense; This is revised with `serverContext.replicator.ActiveTasks()`.